cpp/src/parquet/encoding.cc

```diff
@@ -3149,9 +3160,14 @@ class DeltaByteArrayEncoder : public EncoderImpl, virtual public TypedEncoder<DT
           previous_len = src.len;
           prefix_length_encoder_.Put({static_cast<int32_t>(j)}, 1);
 
-          const uint8_t* suffix_ptr = src.ptr + j;
-          const uint32_t suffix_length = static_cast<uint32_t>(src.len - j);
           last_value_view = view;
+          const auto suffix_length = static_cast<uint32_t>(src.len - j);
+          const uint8_t* suffix_ptr;
+          if (suffix_length == 0) {
+            suffix_ptr = reinterpret_cast<const uint8_t*>("");
```

Comment:
`""` is a temporary variable on stack which only lives in the current scope. What about `suffix_ptr = nullptr` or use a `static const std::string kEmpty = ""`?

Reply:
Indeed! Switched to `nullptr`.